libweasyl/libweasyl/models/tables.py

```diff
@@ -908,6 +908,33 @@ def default_fkey(*args, **kwargs):
 )
 
 
+username_history = Table(
+    'username_history', metadata,
+    Column('historyid', Integer(), primary_key=True, nullable=False),
+    Column('userid', Integer(), ForeignKey('login.userid'), nullable=False),
+    Column('username', String(length=25), nullable=False),
+    Column('login_name', String(length=25), nullable=False),
+    Column('replaced_at', TIMESTAMP(timezone=True), nullable=False),
+    Column('replaced_by', Integer(), ForeignKey('login.userid'), nullable=False),
+    Column('active', Boolean(), nullable=False),
+    Column('deactivated_at', TIMESTAMP(timezone=True), nullable=True),
+    Column('deactivated_by', Integer(), ForeignKey('login.userid'), nullable=True),
+    CheckConstraint("username !~ '[^ -~]' AND username !~ ';'", name='username_history_username_check'),
+    # TODO: replace with generated column once on PostgreSQL 12
+    CheckConstraint("login_name = lower(regexp_replace(username, '[^0-9A-Za-z]', ''))", name='username_history_login_name_check'),
+    CheckConstraint("active = (deactivated_at IS NULL) AND active = (deactivated_by IS NULL)", name='username_history_active_check'),
+)
+
+# enforces one active redirect per user
+Index('ind_username_history_userid', username_history.c.userid, postgresql_where=username_history.c.active, unique=True)
+
+# enforces that active redirects have unique usernames within this table, although they also need to be unique in all of login, logincreate, useralias, and username_history together
+Index('ind_username_history_login_name', username_history.c.login_name, postgresql_where=username_history.c.active, unique=True)
+
+# lookup for a user’s most recent change
```

Comment:
```suggestion
# lookup for a user's most recent change
```
Alembic doesn't like the smart quote here, and it's blocking migration via ``upgrade-db`` when provisioning via a ``SyntaxError`` due to an encoding not being declared. A small portion of the traceback is below.

```
    default: make[1]: Entering directory '/vagrant/libweasyl'
    default: ve/bin/alembic upgrade head
    default: Traceback (most recent call last):
    default:   File "ve/bin/alembic", line 8, in <module>
...
    default: from libweasyl.models.tables import metadata as target_metadata
    default:   File "
    default: /vagrant/libweasyl/libweasyl/models/tables.py
    default: ", line
    default: 934
    default: SyntaxError
    default: :
    default: Non-ASCII character '\xe2' in file /vagrant/libweasyl/libweasyl/models/tables.py on line 934, but no encoding declared; see http://python.org/dev/peps/pep-02
    default: Makefile:44: recipe for target 'upgrade-db' failed
    default: make[1]: Leaving directory '/vagrant/libweasyl'
    default: make[1]: *** [upgrade-db] Error 1
    default: Makefile:67: recipe for target 'upgrade-db' failed
    default: make: *** [upgrade-db] Error 2
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
kyra@kitty ~/g/weasyl> cd libweasyl/libweasyl/models/
kyra@kitty ~/g/w/l/l/models> python
Python 2.7.13 (default, Sep 26 2018, 18:42:22)
[GCC 6.3.0 20170516] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> with open("tables.py") as fp:
...     for i, line in enumerate(fp):
...             if "\xe2" in line:
...                     print i, repr(line)
...
933 '# lookup for a user\xe2\x80\x99s most recent change\n'
```